src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String createManufacturerRequest =
+                "INSERT INTO manufacturers(name,country) values(?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement
+                        = connection.prepareStatement(
+                        createManufacturerRequest, Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet resultSet = createManufacturerStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                return getManufacturerParser(resultSet);
+            } else {
+                throw new RuntimeException("Can't create manufacturer to DB: " + manufacturer);
+            }
```

Comment:
тут не дуже доцільно використовувати метод getManufacturerParser(). Краще повернути до попередньої версії, коли просто діставали згенерований id і все. Помилку також краще видалити.
```suggestion
            if (resultSet.next()) {
                Long id = resultSet.getObject(1, Long.class);
                manufacturer.setId(id);
            }
```